### PR TITLE
DOC : Fix C-API documentation references to items that don't exist

### PR DESCRIPTION
**Pull request by vrindaaa**
Attempting to remove some of the references to items that don't exist. #12632 
- [x] NPY_ITER_USE_MASKNA
- [x] NpyIter_GetMaskNAIndexArray
- [x] NpyMask_Create
- [x] NpyMask_IsExposed
- [x] NpyIter_GetFirstMaskNAOp
- [x] NpyIter_GetMaskNAIndexArray

Edit: added checklist (mattip)

### CODE REVIEW

**Comment by mattip** [comment]
It seems sphinx [released version 2.0.0](https://pypi.org/project/Sphinx/#history) to pypi which broke the document generation ci. See #13206. Once that goes in we can restart the failing CI build. 

Please try to keep your commit message short. You can always add more text to a comment here if more information is needed.

**Comment by vrindaaa** [comment]
Do I need to make any changes here or is this okay?
Also for `NPY_ITER_USE_MASKNA`,` NpyMask_Create` and `NpyMask_IsExposed` which are mentioned in passing under `NpyIter_GetNOp`, `NPY_ITER_ARRAYMASK` and `NPY_ITER_WRITEMASKED` respectively, does the change need to be made in these so they are not mentioned anymore?

**Comment by mattip** [comment]
Looks good so far. Note that `NpyMask_IsExposed` is a little more tricky, it is used in the description of `NPY_ITER_WRITEMASKED` but doesn't actually exist. Since the flag is the C equivalent of [`writemasked`](http://www.numpy.org/devdocs/reference/generated/numpy.nditer.html?highlight=writemasked), the text should be simplified to something like the [H file comment](https://github.com/numpy/numpy/blob/v1.16.2/numpy/core/include/numpy/ndarraytypes.h#L1069) with a reference to writemasked.

**Comment by vrindaaa** [comment]
@mattip references to `NPY_ITER_USE_MASKNA` also exist [here](https://github.com/numpy/numpy/blob/master/doc/neps/nep-0012-missing-data.rst#iterator-mask-features)
So should I remove these also?

File: doc/source/reference/c-api.iterator.rst [hunk 1]
```diff
@@ -603,15 +602,18 @@ Construction and Destruction
...
+            Essentially, this array is the mask for all WRITEMASKED operands.
+            Code uses the WRITEMASKED flag which indicates that only
+            elements where the chosen arraymask operand is True will
+            be written to.
 
             When this flag is used, and this operand is buffered, this
             changes how data is copied from the buffer into the array.
             A masked copying routine is used, which only copies the
-            elements in the buffer for which :c:func:`NpyMask_IsExposed`
+            elements in the buffer for which WRITEMASKED
```
Comment (mattip) [inline doc/source/reference/c-api.iterator.rst]:
```
WRITEMASKED -> ``writemasked``
```
inside double-backticks and the first mention should have a link, something like
```
`writemasked <numpy.nditer>`
```
inside single-backticks so it becomes a link to [here](www.numpy.org/devdocs/reference/generated/numpy.nditer.html)

**Comment by mattip** [comment]
I added a checklist to the top of the PR, please check off the ones you have finished

**Comment by mattip** [comment]
> So should I remove these also?

We don't usually modify NEPS.

File: doc/source/reference/c-api.iterator.rst [hunk 2]
```diff
@@ -603,15 +602,18 @@ Construction and Destruction
...
-            that promise. Code can use the :c:func:`NpyMask_IsExposed`
-            inline function to test whether the mask at a particular
-            element allows writing.
+            this, it is up to the code doing the iteration to follow that
+            promise.
+
+            Essentially, this array is the mask for all `writemasked <numpy.nditer>` operands.
+            Code uses the ``writemasked`` flag which indicates that only
+            elements where the chosen arraymask operand is True will
+            be written to.
```
Comment (mattip) [inline doc/source/reference/c-api.iterator.rst]: text should be limited to 80 columns, you can break on any space even if it is inside a reference

**Comment by mattip** [comment]
Could you [rebase](http://www.numpy.org/devdocs/dev/gitwash/development_workflow.html#rebasing-on-master) on top of master to pick up #13206?

File: doc/source/reference/c-api.iterator.rst [hunk 3]
```diff
@@ -603,15 +602,18 @@ Construction and Destruction
             Indicates that only elements which the operand with
             the ARRAYMASK flag indicates are intended to be modified
             by the iteration. In general, the iterator does not enforce
-            this, it is up to the code doing the iteration to follow
```
Comment (mattip) [inline doc/source/reference/c-api.iterator.rst]: As long as you are touching this, could you clean up the top line? "Indicates that only elements which the operand with the ARRAYMASK flag indicates are intended to be modified by the iteration" does not make sense

**Comment by vrindaaa** [comment]
I don't understand why commit  ENH: Create boolean and integer ufuncs for isnan, isinf, and isfinite is also being shown under my commits? How can I fix that?

**Comment by eric-wieser** [comment]
Can you paste the git commands you ran here? This keeps happening, and it would be nice to work out how it does.

**Comment by vrindaaa** [comment]
These are the commands I used to rebase : 
`git remote add upstream https://github.com/numpy/numpy.git`
`git fetch upstream`
`git checkout issue-12632`
`git rebase upstream/master`
First, rewinding head to replay your work on top of it...
Applying: Fixing Issue #12632 : C-API documentation of nditer lists items that do not exist
Applying: Removing references from C-API documentation which don't exist
Applying: Removing references from C-API documentation which doesn't exist
`git push`
To https://github.com/vrindaaa/numpy.git
 ! [rejected]            issue-12632 -> issue-12632 (non-fast-forward)
error: failed to push some refs to 'https://github.com/vrindaaa/numpy.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
After which I used `git rebase -i HEAD~7` to check the commits and then force-pushed.

**Comment by mattip** [comment]
perhaps you did not `git checkout master; git pull master` before `git checkout issue-12632; git rebase master`.

**Comment by eric-wieser** [comment]
I think using `git rebase -i HEAD~7` to check the commit was the mistake, assuming you actually went through with the rebase

**Comment by vrindaaa** [comment]
I didn't actually went through with the rebase
> perhaps you did not `git checkout master`; `git pull master` before `git checkout issue-12632`; `git rebase master`.

I think it was because I might have forgotten to do this.
Do you think these commands should be added in the documentation [here](http://www.numpy.org/devdocs/dev/gitwash/development_workflow.html#rebasing-on-master)

**Comment by eric-wieser** [comment]
> Edit: after get fetch you need to git merge, git pull does that in one command

Shouldn't have been necessary either, given that there was no need to merge and the intent was to rebase.

Although while we're making one command suggestions, `git pull --rebase upstream/master` would have done the right thing.

**Comment by mattip** [comment]
No, I am wrong. [Eric's comment](https://github.com/numpy/numpy/pull/13205#issuecomment-477885929) is accurate, see the committer name on the extra commit.

**Comment by vrindaaa** [comment]
I should have used `git log` instead. I would keep that in mind next time thank you!

**Comment by vrindaaa** [comment]
How can I remove [this](https://github.com/numpy/numpy/pull/13205/commits/aa100b7179633520126aeffcd2cb544346857c92) as my commit? Even for any new pull requests it is still showing that I commited it.

**Comment by seberg** [comment]
@vrindaaa, I think it should work to remove the commit using `git rebase -i` again. Then do the rebase as Eric suggested using `git pull --rebase upstream master`.

**Comment by vrindaaa** [comment]
I'm sorry I had another trivial doubt. How can I view my documentation after I've made changes in a .rst file?

**Comment by seberg** [comment]
Sorry for missing this a bit. @vrindaaa did you build the documentation by following the steps described here:

http://www.numpy.org/devdocs/docs/howto_build_docs.html

Maybe one tip, to achieve the step:
 > Since large parts of the main documentation are stored in docstrings, you will need to first build NumPy, and install it so that the correct version is imported by

You can run `python runtests.py --shell` which will drop you into a shell which should pick up the correct numpy version. If you set it up differently that is of course also good, but it might be the quickest way to get a started without modifying anything else.

**Comment by vrindaaa** [comment]
Do any changes need to be made here or is this okay?

**Comment by eric-wieser** [comment]
You need to fix the git problems, which is why GitHub is marking this as unmergeable

**Comment by vrindaaa** [comment]
Done!

**Comment by eric-wieser** [comment]
Since your commits all have the same message, you should probably squash them together - also via `rebase -i`

**Comment by mattip** [comment]
Thanks @vrindaaa

**Pull request merged**